docs/build/building-modules/03-msg-services.md

```diff
@@ -98,15 +110,45 @@ The invoked `msgServer` method returns a `proto.Message` response and an `error`
 https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-alpha.0/baseapp/msg_service_router.go#L160
 ```
 
-This method takes care of marshaling the `res` parameter to protobuf and attaching any events on the `ctx.EventManager()` to the `sdk.Result`.
+This method takes care of marshaling the `res` parameter to protobuf and attaching any events on the `EventManager()` to the `sdk.Result`.
 
 ```protobuf reference
 https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-alpha.0/proto/cosmos/base/abci/v1beta1/abci.proto#L93-L113
 ```
 
 This diagram shows a typical structure of a Protobuf `Msg` service, and how the message propagates through the module.
 
-![Transaction flow](https://raw.githubusercontent.com/cosmos/cosmos-sdk/release/v0.46.x/docs/uml/svg/transaction_flow.svg)
+```mermaid
+sequenceDiagram
+    participant User
+    participant baseApp
+    participant router
+    participant handler
+    participant msgServer
+    participant keeper
+    participant EventManager
+
+    User->>baseApp: Transaction Type<Tx>
+    baseApp->>router: Route(ctx, msgRoute)
+    router->>handler: handler
+    handler->>msgServer: Msg<Tx>(Context, Msg(..))
+    
+    alt addresses invalid, denominations wrong, etc.
+        msgServer->>handler: error
+        handler->>router: error
+        router->>baseApp: result, error code
+    else
+        msgServer->>keeper: perform action, update context
+        keeper->>msgServer: results, error code
+        msgServer->>EventManager: Emit relevant events
+        msgServer->>msgServer: maybe wrap results in more structure
+        msgServer->>handler: result, error code
+        handler->>router: result, error code
+        router->>baseApp: result, error code
+    end
+    
+    baseApp->>User: result, error code
+	```
```

Comment:
```Unable to render rich display

Parse error on line 30:
...ult, error code ```## TelemetryNew
----------------------^
Expecting 'SOLID_OPEN_ARROW', 'DOTTED_OPEN_ARROW', 'SOLID_ARROW', 'DOTTED_ARROW', 'SOLID_CROSS', 'DOTTED_CROSS', 'SOLID_POINT', 'DOTTED_POINT', got 'NEWLINE'

For more information, see https://docs.github.com/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams#creating-mermaid-diagrams
```

Reply:
fixed, thanks